modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/components/BadgeFilter/index.js

```diff
@@ -0,0 +1,271 @@
+									minNodes: '',
+								},
+							},
+						}))
+					}
+				/>
+			);
+		},
+		[setFilters]
+	);
+
+	return (
+		<>
+			<div className="d-flex">
+				{!!filters.searchTerm && !loading && (
+					<p className="font-weight-semi-bold m-0 mt-3 text-paragraph-sm">
+						{activationKeysLength} {}
+						result
+						{activationKeysLength > 1 ? 's ' : ' '}
+						for &quot;
+						{filters.searchTerm}&quot;
+					</p>
+				)}
+			</div>
+			<div className="bd-highlight d-flex">
+				<div className="bd-highlight col d-flex flex-wrap pl-0 pt-2 w-100">
+					{!!Object.values(filters.keyType.value).some(
+						(value) => !!value
+					) && <>{getKeyTypeDisplay(filters.keyType)} </>}
+
+					{!!filters.environmentTypes.value?.length && (
+						<BadgePillFilter
+							filterName={filters.environmentTypes.name}
+							filterValue={filters.environmentTypes.value.join(
+								', '
+							)}
+							onClick={() =>
+								setFilters((previousFilters) => ({
+									...previousFilters,
+									environmentTypes: {
+										...previousFilters.environmentTypes,
+										value: [],
+									},
+								}))
+							}
+						/>
+					)}
+
+					{!!filters.instanceSizes.value?.length && (
+						<BadgePillFilter
+							filterName={filters.instanceSizes.name}
+							filterValue={filters.instanceSizes.value.join(', ')}
+							onClick={() =>
+								setFilters((previousFilters) => ({
+									...previousFilters,
+									instanceSizes: {
+										...previousFilters.instanceSizes,
+										value: [],
+									},
+								}))
+							}
+						/>
+					)}
+
+					{!!filters.productVersions.value?.length && (
+						<BadgePillFilter
+							filterName={filters.productVersions.name}
+							filterValue={filters.productVersions.value.join(
+								', '
+							)}
+							onClick={() =>
+								setFilters((previousFilters) => ({
+									...previousFilters,
+									productVersions: {
+										...previousFilters.productVersions,
+										value: [],
+									},
+								}))
+							}
+						/>
+					)}
+
+					{!!filters.status.value?.length && (
+						<BadgePillFilter
+							filterName={filters.status.name}
+							filterValue={filters.status.value.join(', ')}
+							onClick={() =>
+								setFilters((previousFilters) => ({
+									...previousFilters,
+									status: {
+										...previousFilters.status,
+										value: [],
+									},
+								}))
+							}
+						/>
+					)}
+
+					{!!(
+						filters.expirationDate.value.onOrAfter ||
+						filters.expirationDate.value.onOrBefore
+					) && (
+						<BadgePillFilter
+							filterName={filters.expirationDate.name}
+							filterValue={getDatesDisplay(
+								filters.expirationDate
+							)}
+							onClick={() =>
+								setFilters((previousFilters) => ({
+									...previousFilters,
+									expirationDate: {
+										...previousFilters.expirationDate,
+										value: {
+											onOrAfter: undefined,
+											onOrBefore: undefined,
```

Comment:
Should he change from undefined to false here too? @iagochaves 
```suggestion
											onOrAfter: false,
											onOrBefore: false,
```